featuretools/tests/synthesis/test_encode_features.py

```diff
@@ -154,3 +153,22 @@ def test_encode_features_drop_first():
                                                  include_unknown=False)
 
     assert len(features_enc.columns) == 2
+
+
+def test_encode_features_handles_dictionary_input(es):
+    f1 = IdentityFeature(es["log"]["product_id"])
+    f2 = IdentityFeature(es["log"]["purchased"])
+    f3 = IdentityFeature(es["log"]["session_id"])
+
+    features = [f1, f2, f3]
+    feature_matrix = calculate_feature_matrix(features, es, instance_ids=[0, 1, 2, 3, 4, 5])
+    top_n_dict = {f1: 2, f3: 1}
+    feature_matrix_encoded, features_encoded = encode_features(feature_matrix, features, top_n=top_n_dict)
+    assert len(features_encoded) == 6
+
+    feature_matrix_encoded, features_encoded = encode_features(feature_matrix, features, top_n=top_n_dict, include_unknown=False)
+    assert len(features_encoded) == 4
```

Comment:
for these tests, instead of checking the number of features can we check against a list of expected feature names